### DISCUSSION TITLE
Show the dependencies of a variable

### DISCUSSION THREAD

**Discussion by AntoineThebaud** [perses/perses#878] [Ideas]
The idea is to be able to see which other variable(s) a given variable depends on when you don't have the right to edit a dashboard.

We could imagine having a simple popup displayed above the variable on hover / right-click / etc, showing a coma-separated list of the variables required to build the considered one.

**Comment by AntoineThebaud** [comment]
Another similar use case: it would also be good to have the ability to see for a given panel on which variables it depends.

**Comment by sjcobb** [comment]
Definitely agree this would be great to have. The panel-level part of this relates to some of the proposed changes in [this discussion](https://github.com/perses/perses/discussions/956), if we move where queries are defined up a level so it is part of the top-level panel spec, this will be a lot easier to implement! I know Mohit has some ideas too that he can create mockups for to get feedback. 

We particularly need the ability to see which template variables are used on a panel if we decide not to support variables being used in a panel’s `name`

**Comment by AntoineThebaud** [comment]
Created issue https://github.com/perses/perses/issues/1539 to follow-up on this one.
Please use this issue to further discuss the proposal.
Closing the discussion accordingly.
